### DISCUSSION TITLE
Run items not appearing when linking to a trace and not a span or a generation

### DISCUSSION THREAD

**Discussion by ladislasdellinger** [langfuse/langfuse#1357] [Support] [answered]
Hello,

I noticed that when linking a dataset run item to a trace and not a span or a generation, it doesn't appears later in the dataset run items table.

Is this a wanted behaviour or is this a bug ?

**Comment by marcklingen** [comment]
This is currently unsupported but on the short-term backlog to add

**Reply by ladislasdellinger** [reply]
Okay !
I would be pleased to make a PR to add this also. However, I'm not sure what would be the good way to do it knowing that currently a `run_item` has an `observation_id` (which can't be a trace then) 

Do you have any idea what would be the best/cleanest way to add this ?

**Comment by marcklingen** [comment, marked as answer]
@ladislasdellinger, work in progress pr: #1502 (just fyi)

**Reply by ladislasdellinger** [reply]
Hi @marcklingen ! Thanks for the update :)

I also had a PR for this but it was a WIP too => #1438 

I don't know if there is something useful in my PR. If not, we can close it to avoid duplicates

**Reply by marcklingen** [reply]
sorry, I overlooked this pr and did not merge fast enough / provide feedback. will get this merged today (#1502), thank you for the inspiration!

**Reply by marcklingen** [reply]
side note, it would be nice if GH discussions would display the linked issues on a discussion thread

**Reply by ladislasdellinger** [reply]
No problem @marcklingen! Glad if my PR has help in any way tho ;) 

Totally agree for the side note 😅

Tell me if you have any question on my PR!
